File: java/src/main/java/api/cornell/ClassLevelsResponse.java
package api.cornell;

import api.cornell.data.classes.ClassLevel;
import org.jetbrains.annotations.Nullable;

import java.util.List;

/**
 * {@code ClassLevelsResponse} is the response of a Class Levels request.
 */
final class ClassLevelsResponse {
    
    /**
     * Main data of response.
     */
    @Nullable
    private final Data data;
    
    /**
     * Constructor for GSON.
     */
    private ClassLevelsResponse() {
        data = null;
    }
    
    /**
     * Obtain the class levels information.
     *
     * @return the class levels information.
     */
    @Nullable
    public List<ClassLevel> getClassLevels() {
        if (data == null) {
            return null;
        }
        return data.classLevels;
    }
    
    /**
     * {@code Data} is an uninteresting class that just holds an array of {@code Page}.
     */
    private static class Data {
        
        @Nullable
        private List<ClassLevel> classLevels;
        
        /**
         * Constructor for GSON.
         */
        private Data() {
            classLevels = null;
        }
        
    }
}

File: kotlin/src/main/kotlin/api/cornell/response/classes/ClassLevelsResponse.kt
package api.cornell.response.classes

import api.cornell.data.classes.ClassLevel

/**
 * [ClassLevelsResponse] is the response of a Class Levels request.
 */
internal class ClassLevelsResponse private constructor() {

    /**
     * Main data of response.
     */
    private val data: Data? = null
    /**
     * Obtain the class levels information.
     */
    val classLevels: List<ClassLevel>?
        get() = data?.classLevels

    /**
     * [Data] is an uninteresting class that just holds an array of [ClassLevel].
     */
    private inner class Data private constructor() {
        val classLevels: List<ClassLevel>? = null
    }

}